src/lib/filehandler.rs

```diff
@@ -1,16 +1,52 @@
-use std::fs::File;
+use std::fs::*;
+use std::io::*;
+use basic_types::instruction_set::*;
+use basic_types::instruction::*;
+#[derive(Debug)]
 pub struct FileHandler {
     path: String,
-    file: File,
+    buf: BufReader<File>,
 }
 impl FileHandler {
     pub fn new(path: String) -> FileHandler {
         let file = File::open(&path).unwrap();
+        let f = BufReader::new(file);
         return FileHandler {
             path: path,
-            file: file,
+            buf: f,
         };
     }
+    pub fn read_instruction(&mut self) -> Option<Result<String>> {
```

Comment:
Why don't you just return an `Option<Result,&str>` instead of `Option<Result>` ?

the `&str` might hold the error message, if you want to skip it, just use an option